index.js

```diff
@@ -224,20 +209,33 @@ class Analytics {
 
     debug('flush: %o', data)
 
-    request
-      .post(`${this.host}/v1/batch`)
-      .auth(this.writeKey, '')
-      .timeout(this.timeout)
-      .retry(this.retryCount)
-      .send(data)
-      .end((err, res) => {
-        err = err || error(res)
+    const done = err => {
+      callbacks.forEach(callback => callback(err))
+      callback(err, data)
 
-        callbacks.forEach(callback => callback(err))
-        callback(err, data)
+      debug('flushed: %o', data)
+    }
+
+    const req = {
+      method: 'POST',
+      url: `${this.host}/v1/batch`,
+      data
+    }
+
+    if (this.timeout) {
+      req.timeout = typeof this.timeout === 'string' ? ms(this.timeout) : this.timeout
+    }
+
+    axios(req)
+    .then(() => done())
+    .catch(err => {
+      if (err.response) {
+        const error = new Error(err.response.statusText)
```

Comment:
=> `err = new Error(err.response.statusText)`

Reply:
Maybe it’s just that I’ve been writing a lot of Golang lately (and `govet` yells), but I think shadowing the existing variable decreases readability. I’m open to changing this if you have a strong opinion here tho

Reply:
I hate overwriting variables either, const all the way. Just thought that in this block it wouldn't matter. It can stay that way too, not a big deal.